pkg/controller/subjectpermission/subjectpermission_controller.go

```diff
@@ -133,30 +132,27 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
 	}
 
 	if len(instance.Spec.ClusterPermissions) != 0 {
-		// build a clusterRoleBindingNameList which consists of clusterRoleName-subjectName
-		crClusterRoleBindingNameList := buildClusterRoleBindingCRList(instance)
-
-		// check ClusterRoleBindingName
-		populateCrClusterRoleBindingNameList := populateClusterRoleBindingNames(crClusterRoleBindingNameList, clusterRoleBindingList)
-
 		var CreatedClusterRoleBindingCount int
-		// loop through crClusterRoleBindingNameList
-		for _, clusterRoleBindingName := range populateCrClusterRoleBindingNameList {
-			// get the clusterRoleName by spliting the clusterRoleBindng name
-			clusterRBName := strings.Split(clusterRoleBindingName, "-")
-			clusterRoleName := clusterRBName[0]
-			subjectName := clusterRBName[1]
+		// build a clusterRoleNameList
+		var crClusterRoleNameList []string
+		for _, clusterRoleName := range instance.Spec.ClusterPermissions {
```

Comment:
why are you creating a slice out of a slice? You should just range over this slice directly.